auth-api/src/auth_api/services/org.py

```diff
@@ -227,65 +228,84 @@ def _create_payment_settings(
-            response = RestService.put(
-                endpoint=f"{pay_url}/accounts/{org_model.id}", data=pay_request, token=token, raise_for_status=True
-            )
-
-        if response.status_code == HTTPStatus.OK:
-            payment_account_status = PaymentAccountStatus.CREATED
-        elif response.status_code == HTTPStatus.ACCEPTED:
-            payment_account_status = PaymentAccountStatus.PENDING
-        else:
-            payment_account_status = PaymentAccountStatus.FAILED
+            if payment_method == PaymentMethod.PAD.value:  # PAD has bank related details
+                pay_request["paymentInfo"]["bankTransitNumber"] = payment_info.get("bankTransitNumber", None)
+                pay_request["paymentInfo"]["bankInstitutionNumber"] = payment_info.get("bankInstitutionNumber", None)
+                pay_request["paymentInfo"]["bankAccountNumber"] = payment_info.get("bankAccountNumber", None)
+                pay_request["padTosAcceptedBy"] = kwargs["user_context"].user_name
+            # invoke pay-api
+            token = RestService.get_service_account_token()
+            if is_new_org:
+                response = RestService.post(
+                    endpoint=f"{pay_url}/accounts", data=pay_request, token=token, raise_for_status=True
+                )
+            else:
+                response = RestService.put(
+                    endpoint=f"{pay_url}/accounts/{org_model.id}", data=pay_request, token=token, raise_for_status=True
+                )
 
-        if payment_account_status != PaymentAccountStatus.FAILED and payment_method:
-            payment_method_description = (
-                PaymentMethod(payment_method).name if payment_method in [item.value for item in PaymentMethod] else ""
-            )
-            ActivityLogPublisher.publish_activity(
-                Activity(
-                    org_model.id,
-                    ActivityAction.PAYMENT_INFO_CHANGE.value,
-                    name=org_model.name,
-                    value=payment_method_description,
+            if response.status_code == HTTPStatus.OK:
+                payment_account_status = PaymentAccountStatus.CREATED
+            elif response.status_code == HTTPStatus.ACCEPTED:
+                payment_account_status = PaymentAccountStatus.PENDING
+            elif response.status_code == HTTPStatus.CREATED:
+                payment_account_status = PaymentAccountStatus.FAILED
+            else:
+                payment_account_status = PaymentAccountStatus.FAILED
+                if response.json:
+                    error_payload = response.json()
+                    error_code = error_payload.get("error", "UNKNOWN_ERROR")
+                else:
+                    error_code = "UNKNOWN_ERROR"
+                logger.error(f"Account create payment Error: {response.text}")
+
+            if payment_account_status != PaymentAccountStatus.FAILED and payment_method:
+                payment_method_description = (
```

Comment:
So it's possible to be "failed" but still have your payment account updated and skip the activity log?

Reply:
I'm not quite sure on this one, I thought it might be throw exception and tried, but too many unit tests failed

Reply:
I'm pretty sure even though it "FAILED" if the response was 200, it still was updated in sbc-pay - the payment method would still be changed then, you'll have to recreate the scenario and watch the activity log?
Unless it's possible it's because of the PAD confirmation period?

Anyways might not be worth looking into, if we can just keep the same flow as before.

Reply:
yes I keep the same flow as before